lib/ansible/module_utils/satellite.py

```diff
@@ -0,0 +1,84 @@
+# (c) 2018, Luc Stroobant (luc.stroobant@wdc.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+try:
+    import requests
+    from requests.auth import HTTPBasicAuth
+    requests.packages.urllib3.disable_warnings()
+    requests_available = True
```

Comment:
Ah, on that. I started with the built-in url modules at first. It has a serious performance issue, are you aware of that? I found out when running my Satellite reporting module (which executes 3-4000 requests to fetch info about all our hosts).
For every request you make, the module creates a temp file where it creates a single file with a bundle of all CA's on your system (about 1.4MB big in my case). After 1024 requests the process is killed since the user can not open any more files. It's here in the code:
https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/urls.py#L710

Since all tasks are delegated to the Ansible controller and requests also makes some other things (like the management of the params) easier, I think it should be acceptable to use it here.

Reply:
Whoaw, I knew there was an impact, but did not suspect this. This is ugly.
cc @kbreit